src/main/java/in/projecteka/consentmanager/consent/ConsentManager.java

```diff
@@ -66,408 +71,458 @@
+                                .detail(requestedDetail)
+                                .id(requestId)
+                                .build()))
+                        .thenReturn(requestId.toString()));
+    }
+
+    public Mono<Void> requestConsent(RequestedDetail requestedDetail, UUID requestId) {
+        return Mono.just(requestId)
+                .filterWhen(this::validateRequest)
+                .switchIfEmpty(Mono.error(ClientError.requestAlreadyExists()))
+                .flatMap(val -> validatePatient(requestedDetail.getPatient().getId())
+                        .then(validatePurpose(requestedDetail.getPurpose()))
+                        .then(validateHiTypes(requestedDetail.getHiTypes()))
+                        .then(validateHIPAndHIU(requestedDetail)))
+                .flatMap(r -> Mono.defer(() -> saveConsentRequest(requestedDetail, requestId)));
+    }
+
+    private Mono<Void> saveConsentRequest(RequestedDetail requestedDetail, UUID requestId) {
+        ConsentRequestId request = ConsentRequestId.builder()
+                .id(requestId)
+                .build();
+        ConsentRequestResult consentRequestResult = ConsentRequestResult.builder()
+                .requestId(requestId)
+                .timestamp(Instant.now().toString())
+                .consentRequest(request)
+                .resp(GatewayResponse.builder().requestId(UUID.randomUUID().toString()).build())
+                .build();
+
+        return consentRequestRepository.insert(requestedDetail, requestId)
+                .doOnSuccess(s -> Mono.defer(() -> consentManagerClient
+                        .sendInitResponseToGateway(consentRequestResult, requestedDetail.getHiu().getId())).subscribe());
+    }
+
+    private Mono<Boolean> validateRequest(UUID requestId) {
+        return consentRequestRepository.requestOf(requestId.toString())
+                .map(Objects::isNull)
+                .switchIfEmpty(Mono.just(true));
+    }
+
+    private Mono<Void> validatePurpose(ConsentPurpose purpose) {
+        return conceptValidator.validatePurpose(purpose.getCode())
+                .flatMap(result ->
+                        result.booleanValue() ? Mono.empty()
+                                : Mono.error(new ClientError(BAD_REQUEST,
+                                new ErrorRepresentation(new Error(INVALID_PURPOSE, "Invalid Purpose"))))
+                );
+    }
+
+    private Mono<Void> validateHiTypes(HIType[] hiTypes) {
+        return conceptValidator.validateHITypes(Arrays.stream(hiTypes).map(type -> type.getValue()).collect(Collectors.toList()))
+                .flatMap(result ->
+                        result.booleanValue() ? Mono.empty()
+                                : Mono.error(new ClientError(BAD_REQUEST,
+                                new ErrorRepresentation(new Error(INVALID_HITYPE, "Invalid HI Type"))))
+                );
+    }
+
```

Comment:
Whatever refactoring I had done is all gone when you rebased with your branch. Can you fix that? Check master?

Not just this method, there are other methods too.

Reply:
Fixed. Something went wrong on rebasing. All the methods are up to date with master now.
Thank you